Libraries/libbom/Sources/bom.c

```diff
@@ -334,11 +334,11 @@ bom_variable_iterate(struct bom_context *context, bom_variable_iterator iterator
     context->iteration_count++;
 
     struct bom_header *header = (struct bom_header *)context->memory.data;
-    struct bom_variables *vars = (struct bom_variables *)((void *)header + ntohl(header->variables_offset));
+    struct bom_variables *vars = (struct bom_variables *)((uintptr_t)header + ntohl(header->variables_offset));
 
     ptrdiff_t var_offset = 0;
     for (size_t i = 0; i < ntohl(vars->count); i++) {
-        struct bom_variable *var = (struct bom_variable *)((void *)vars->first + var_offset);
+        struct bom_variable *var = (struct bom_variable *)((uintptr_t)vars + sizeof(struct bom_variable) + var_offset);
```

Comment:
I had to remove the `first` array because it ended up with nested zero-length arrays, and 
MSVC complained about that. Since it was the last item in the `struct bom_variable`, adding the sizeof is equivalent to the old array location.